components/brave_ads/core/internal/common/url/url_util_internal.cc

```diff
@@ -0,0 +1,115 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_ads/core/internal/common/url/url_util_internal.h"
+
+#include <string>
+
+#include "base/containers/contains.h"
+#include "net/base/registry_controlled_domains/registry_controlled_domain.h"
+#include "net/base/url_util.h"
+#include "url/gurl.h"
+
+namespace brave_ads::internal {
+
+namespace {
+
+constexpr char kBraveScheme[] = "brave";
+constexpr char kChromeScheme[] = "chrome";
+
+constexpr char kRewardsHostName[] = "rewards";
+constexpr char kSyncHostName[] = "sync";
+constexpr char kWalletHostName[] = "wallet";
+
+constexpr char kSettingsHostName[] = "settings";
+
+constexpr char kSearchEnginesPath[] = "/searchEngines";
+constexpr char kSearchPath[] = "/search";
+constexpr char kSearchQuery[] = "search";
+
+}  // namespace
+
+GURL ReplaceBraveSchemeWithChromeScheme(const GURL& url) {
+  CHECK(url.is_valid());
+
+  if (!url.SchemeIs(kBraveScheme)) {
+    return url;
+  }
+
+  GURL::Replacements replacements;
+  replacements.SetSchemeStr(kChromeScheme);
+  return url.ReplaceComponents(replacements);
+}
+
+bool HasUrlQueryNameWithNonEmptyValue(const GURL& url,
```

Comment:
Perhaps this should be `HasOnlyQueryNameWithNonEmptyValue` as it matches the logic of the function better.

Reply:
We iterate through all the query names and values, so it won't be the only match.

Reply:
Fixed